src/libs/sfdk/dockervirtualmachine_p.h

```diff
@@ -0,0 +1,96 @@
+** conditions contained in a signed written agreement between you and
+** The Qt Company.
+**
+****************************************************************************/
+#ifndef DOCKERCONTAINER_P_H
+#define DOCKERCONTAINER_P_H
+
+#include "virtualmachine_p.h"
+#include "dockervirtualmachine.h"
+
+namespace Sfdk {
+
+class DockerVirtualMachinePrivate : public VirtualMachinePrivate
+{
+    Q_DECLARE_PUBLIC(DockerVirtualMachine)
+
+public:
+    using VirtualMachinePrivate::VirtualMachinePrivate;
+    virtual ~DockerVirtualMachinePrivate();
+
+    QVariantMap toMap() const;
+
+
+    void fetchInfo(VirtualMachineInfo::ExtraInfos extraInfo, const QObject *context,
+                   const Functor<const VirtualMachineInfo &, bool> &functor) const;
+
+    void start(const QObject *context, const Functor<bool> &functor);
+    void stop(const QObject *context, const Functor<bool> &functor);
+    void probe(const QObject *context,
+               const Functor<BasicState, bool> &functor) const;
+
+    void setVideoMode(const QSize &size, int depth, const QString &deviceModelName,
+            Qt::Orientation orientation, int scaleDownFactor, const QObject *context,
+            const Functor<bool> &functor) override;
+
+protected:
+    void doSetMemorySizeMb(int memorySizeMb, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetCpuCount(int cpuCount, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetStorageSizeMb(int storageSizeMb, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetSharedPath(SharedPath which, const Utils::FileName &path, const QObject *context,
+            const Functor<bool> &functor) override;
+
+    void doAddPortForwarding(const QString &ruleName,
+        const QString &protocol, quint16 hostPort, quint16 emulatorVmPort,
+        const QObject *context, const Functor<bool> &functor) override;
+    void doRemovePortForwarding(const QString &ruleName,
+        const QObject *context, const Functor<bool> &functor) override;
+    void doSetReservedPortForwarding(ReservedPort which, quint16 port,
+            const QObject *context, const Functor<bool> &functor) override;
+    void doSetReservedPortListForwarding(ReservedPortList which, const QList<Utils::Port> &ports,
+            const QObject *context, const Functor<const QMap<QString, quint16> &, bool> &functor)
+        override;
+
+    void doRestoreSnapshot(const QString &snapshotName, const QObject *context,
+        const Functor<bool> &functor) override;
+
+private:
+    void doSetReservedPortForwarding(ReservedPort which, quint16 port);
+
+    void setSharedPathFromJSon(SharedPath which, const QJsonObject& json, VirtualMachineInfo& info);
+    VirtualMachineInfo virtualMachineInfoFromOutput(const QByteArray& output) const;
+
+    static QString label(SharedPath which);
+    static QString label(ReservedPort which);
```

Comment:
I would prefer naming it `labelFor` (the argument is not of a type dedicated to select a label).

Reply:
fixed